crates/katana/primitives/src/genesis/json.rs

```diff
@@ -530,8 +629,30 @@ pub fn to_base64(genesis: GenesisJson) -> Result<Vec<u8>, GenesisJsonError> {
 
 /// Deserialize the [GenesisJson] from base64 encoded bytes.
 pub fn from_base64(data: &[u8]) -> Result<GenesisJson, GenesisJsonError> {
+    // Decode base64 bytes
     let decoded = BASE64_STANDARD.decode(data)?;
-    Ok(serde_json::from_slice::<GenesisJson>(&decoded)?)
+    // Deserialize JSON
+    let mut genesis_json: GenesisJson = serde_json::from_slice(&decoded)?;
+
+    // Populate name field
```

Comment:
What do you want to achieve here? The name of the class is supposed to be given by the user into the genesis json file. :) We don't need to retrieve it from the artifact file itself.

Reply:
OK. I have removed the for loop